In [72]:
# libraries
import numpy as np
import pandas as pd
import altair as alt

df = pd.read_csv('../data/edattainxtry_5plus.csv')

# PSTAT 100 Project plan

This is a guide to preparing your project plan. It functions both as a guide to the work you'll need to do and as a guide to preparing the deliverable. You can use it as a template to draft the plan report; if so, **please remove the text explanations and instructions in each section so that it reads as a coherent and continuous document**.

While you may find it useful initially to follow the outline given, you do not need to adhere to it exactly -- you're free to organize your submission in the way that seems most natural to you. However, please do keep the high-level sections, so that your report includes the following headers:

0. Background
1. Data description
2. Initial exporations
3. Planned work

Your report does not need to be long. It should be about 2-4 pages, and might not be much longer than this template once you replace the guiding text with your own work.

## Group information

**Group members**:

Yibo Liang

Aarya Kulkarni

Alan Su

Nicole Magallanes Flores


**Contributions**:
1. Alan Su: worked on finding data and data background (Part 0)
2. Yibo Liang: worked on tidying the dataset/initial exploration (Part 2)
3. Aarya Kulkarni: worked on data description (Part 1)
4. Nicole Magallanes Flores: worked on planned work (Part 3)

---
## 0. Background

This section should introduce your reader to the general topic you're engaging with in your project and explain any specialized knowledge that they may need to understand your dataset and why it's interesting. It doesn't need to be long, but should touch on the following points:
* Introduce the topic of your project.
* What area or areas of study are you in dialogue with for your project?
* What is your data about, broadly? 
* What is the motivation for collecting the kind of data you're working with, and what sorts of things could you potentially learn?

You can look to the background sections in the homework assignments for examples. (There you can also see how to include images in your notebook.) The background sections of the homeworks are usually short and focused paragraphs intended to orient you to what you'll do in the assignment. They don't go into a lot of detail -- just enough to (hopefully) convince you that the data are interesting and explain any terminology or general information you may not know.

You may find it useful to write up the data description first, think about what the reader should know before they peek at your dataset, and then come back to the background section. I often write the background sections of your assignments last, once I have a sense of what kind of information would be most useful going into the assignment.

### Background

Education is an important issue in every country around the world, as it is with education that people can innovate and drive their country forward. But while education is obviously important to many people, not everyone is able to have access to an education. Some countries may not have the resources to put every citizen through school, or some countries may have the resources, but for example, people that live in rural areas might choose not to go through education in favor of continuing a family business or pursuing their own economic endeavors.

The dataset that we have chosen to work with is a dataset taken from [The World Bank's data catalog](https://datacatalog.worldbank.org/search/dataset/0038973/Education-Attainment-and-Enrollment-around-the-World), their dataset on educational attainment and enrollment around the world. This dataset takes data from many types of surveys from all around the world in order to get data on things such as socioeconomic conditions, or living standards such as rural or urban. Using this data, we will explore how educational attainment is affected by conditions such as wealth, gender, and living location, and can use this information to advocate for better educational attainment standards everywhere in the world. Is the world bank right when they say that there are large differences in educational attainment by wealth? Will gender and living standards affect a person's educational attainment? Using this data, we plan on looking into these questions and hope to gain a better understanding of what educational attainment is like around the world.


---
## 1. Data description

This section should introduce your dataset in detail. It should reflect your having gone through the collect/acquaint/tidy stages of the lifecycle. Below I've provided you with an outline. You do not need to adhere to this strictly -- in fact, it would be more natural to divide the items among a few short paragraphs -- but you should touch on each item in a format that suits your project.

### Basic information

Help your reader understand what your data is, where it came from, and how it can be used. Provide the following.

**General description**: provide a one- or two-sentence description of the data right at the beginning. For instance, "The data are diatom counts sampled from evenly-spaced depths in a sediment core from the gulf of California." Nothing too complicated, just something to give your reader a sense of the 'what' right off the bat.

**Source**: indicate where your data came from. Provide a verbal description -- who collected it as part of what project and where -- and either a citation or a hyperlink.

**Collection methods**: How were the data values obtained? Provide a simple description of how measurements were taken (using scientific equipment? web scraping? surveys?).

**Sampling design and scope of inference**: Indicate the relevant population. If identifiable from data documentation, state the sampling frame and sampling mechanism and indicate the scope of inference. If no information is available about the sampling design, indicate this instead, and discuss the extent to which having no scope of inference is a limitation for the particular topic you're investigating.

The data describes the percentage of population ages 15 to 19 that has completed each grade (1-9) in developing countries around the world spanning multiple years (1990-2020). 

This data came from a World Bank database with summary information on education level attained, taken from household surveys from developing countries around the world (http://www.worldbank.org/en/research/brief/edattain). Household-based surveys used to create this dataset include: Demographic and Health Surveys (DHS- http://www.measuredhs.com), Multiple Indicator Cluster Surveys (MICS- http://www.childinfo.org), Living Standards Measurements Study Surveys (LSMS- http://www.worldbank.org/lsms), and other household-based surveys (country specific), ex: socio-economic surveys. In addition to these household-based surveys, selected country/year variables were added to the dataset from the World Development Indicators database (http://databank.worldbank.org). Some variables added include gdp per capita (based on 2015 U.S. dollars) and the consumer price index (based on 2010) for various countries. 

The relevant population is the population ages 15-19 in the countries surveyed. Since the dataset is aggregated from multiple surveys. Some documentation was provided on two of the types of household surveys performed. 

The Demographic and Health Surveys (DHS) used a mix of questionnaires, biomarkers, and geographic information as survey tools to conduct this survey. This survey's sampling design consisted of a two-stage stratified cluster design in which Enumeration Areas (areas canvased) were drawn from Census files (stage 1), followed by a sample of households being drawn from a list in each Enumeration Area (stage 2). 

The Living Standards Measurement Study Surveys (LSMS)'s sample frame is given by the Population and Housing Census. Following this, a two round sampling method is used. The first round selects Primary Selection Units (PSU) through random sampling. The second round selects subunits from each PSU using a method of systematic sampling. 

Since the sampling frame is usually given by the country's Census, and the sampling mechanism involved random sampling, we can say that the scope of inference are the households captured in the Census.

### Data semantics and structure

**Units and observations**: State the observational units.

The observational units are countries around the world.

**Variable descriptions**: Provide a table of variable descriptions. If your dataset is large and you'll only work with a subset of the total available variables, limit your attention to the variables that you'll work with. Here's a template you can work with:

Name | Variable description | Type | Units of measurement
---|---|---|---
country | observtion country | Character | Name of Country 
year | year of observation | Numeric | Calendar year 
gdppc2015 | gdp per capita (2015 U.S. dollars) | Numeric | gdppc 
cpi2010 | consumer price index (Based on 2010) | Numeric | cpi(hundreds) 
level | level of education| Numeric | grade 
sex | sex of group | Factor | male/female 
location | location development | Factor | urban/rural 
prop| proportion of the sample that attained the education | Numeric | percentage 


**Example rows**: Print a few example rows of your dataset in tidy format. Please don't include the codes you used to manipulate the raw data. Do that in a separate notebook and export the result to a .csv file -- `data.to_csv('tidy-data.csv')` -- to load directly into the cell below.

In [81]:
# load tidied data and print rows
data_tidy = pd.read_csv('../data/tidy_data.csv').drop(columns='Unnamed: 0')
data_tidy.head(10)

country  year  gdppc2015  cpi2010 location   sex  level   prop
0  Afghanistan  2015    556.007  132.883    Urban  Male      1  0.926
1  Afghanistan  2007    392.710   83.074    Urban  Male      1  0.846
2  Afghanistan  2010    526.104  100.000    Urban  Male      1  0.862
3       Angola  2015   4166.980  159.405    Urban  Male      1  0.976
4       Angola  1999   2458.096    0.684    Urban  Male      1    NaN
5       Angola  2008   4077.777   76.813    Urban  Male      1  0.922
6       Angola  2001   2471.665    7.347    Urban  Male      1  0.921
7      Albania  2002   2247.498   82.382    Urban  Male      1  0.968
8      Albania  2017   4249.804  115.059    Urban  Male      1  0.987
9      Albania  2000   1960.882   74.138    Urban  Male      1  0.984

---
## 2. Initial explorations

At this stage, you may spend most of your effort on the computing side tidying up the data. You're not expected to complete a thorough exploratory analysis, and if your dataset was especially messy to start with, you may not even begin your exploratory analysis by the time you prepare this report. You have the option to leave exploration for the next stage of work and simply report basic properties of the dataset, but you should at minimum address the items in the 'basic properties' section below.

### Basic properties of the dataset

Help the reader get acquainted with your dataset on a simple level by identifying characteristics of the dataset and variable summaries. Some amount of code is fine here, but try to use code cells sparingly.

**Dimensions**: state the dimensions of the data (in tidy format, of course).

**Missing values**: Are there missing values? If so, why are they missing?

**Variable summaries**: Provide simple variable summaries for the most important variables in your dataset. Preferably, you'll do this for all variables, but if you have a large number, you might need to prioritize and focus on the ones most of interest. What exactly you do is a little case-specific, but think of things like means and variances, min/max, number of levels and observation counts for categorical variables, etc.

### Dimensions

In [64]:
data_tidy.shape

(22536, 8)

The tidy data has 22536 rows and 8 columns.

### Missing Data

In [65]:
data_tidy.isna().mean()

country      0.000000
year         0.000000
gdppc2015    0.011182
cpi2010      0.070288
location     0.000000
sex          0.000000
level        0.000000
prop         0.015974
dtype: float64

There are so missingness in our tidy data. Specifically, `gdppc2015` is missing about 1.1% of values, `value` is missing about 1.6% of its values, and `cpi2010` is missing about 7% of its values. This data is missing because the country did not report its gdp/cpi that year.

One interesting thing of note is that year is not continuous for all countries. Some countries did not report anything for certain years; therefore, we need to keep track of these gap years.

### Variables Summary

Alot of our variables are binary factors; so we will take a look at the most important variables of interest: `gdppc2015`, `cpi2010`, and `prop`. Furthermore, these summaries will be grouped by years as it does not make sense to look at the summary for all the years combined.

#### `gdppc2015` and `cpi2010` summary grouped by year.

In [79]:
data_tidy.loc[:, ['year','gdppc2015']].groupby('year').describe()

gdppc2015                                                              \
         count         mean           std       min         25%        50%   
year                                                                         
1981      36.0  6079.244000  9.223960e-13  6079.244  6079.24400  6079.2440   
1985      36.0  6178.524000  9.223960e-13  6178.524  6178.52400  6178.5240   
1989     108.0  4700.474667  2.758735e+03   883.763   883.76300  5988.5410   
1990     288.0  3790.076875  1.541082e+03   924.635  3075.92400  3759.5450   
1991     360.0  2609.108600  1.732949e+03   553.526  1563.29500  2015.6735   
1992     504.0  1091.065857  8.656860e+02   243.371   457.65300   696.7560   
1993     396.0  2070.116727  1.624154e+03   495.764   937.44600  1643.1490   
1994     468.0  2402.668769  2.118550e+03   355.148   576.68900  1662.9130   
1995     576.0  3375.670375  1.975943e+03   243.598  1886.00875  3909.6910   
1996     720.0  2724.908650  2.473522e+03   236.461   571.92200  1698.7890   
1997     504.0  1692.322286  1.676927e+03   256.173   409.84700  1003.5640   
1998     864.0  2142.724250  2.039718e+03   322.152   737.11075  1257.6285   
1999     504.0  2281.404571  1.534548e+03   444.323   633.18300  2108.5465   
2000    2052.0  2191.982509  2.449140e+03   262.025   505.50100  1276.7600   
2001     792.0  2274.697955  2.003909e+03   334.120   925.71600  1377.5440   
2002     648.0  2765.765333  1.822859e+03   808.440  1679.40600  2156.8700   
2003     684.0  1964.514895  2.213493e+03   353.461   484.09700  1352.9790   
2004     540.0  1772.718333  1.921918e+03   287.272   624.60400  1129.1370   
2005    1512.0  2344.778619  2.109214e+03   289.047   696.33900  1569.2405   
2006    1332.0  2529.748514  2.222323e+03   294.520   809.47400  1575.1430   
2007     720.0  1529.747800  1.188397e+03   381.456   710.64125   975.3675   
2008     576.0  1976.688125  1.946544e+03   447.684   779.23525  1378.9770   
2009     648.0  3933.197889  2.636634e+03   553.517  1630.08900  3242.9480   
2010    1116.0  1766.866323  2.182464e+03   313.312   531.71300   808.1410   
2011     900.0  2744.871560  2.333917e+03   436.550   844.90600  2144.7000   
2012     864.0  3675.431333  3.169458e+03   464.148  1129.20550  2585.8795   
2013     576.0  3476.779187  3.584188e+03   370.301   726.26875  2240.1590   
2014     900.0  2266.863760  1.667194e+03   583.566  1104.75000  1430.4450   
2015     576.0  3720.502063  3.150940e+03   380.597  1134.54050  2606.4470   
2016     468.0  2235.418308  2.420078e+03   294.219   830.24600  1351.9240   
2017     324.0  2188.801556  1.206894e+03   622.530  1112.81700  1896.3670   
2018     792.0  3180.022182  3.128527e+03   412.990   802.03400  1531.6560   
2019     828.0  4536.952696  5.816438e+03   649.760   885.60000  2482.1200   
2020      72.0  1680.820500  1.317819e+02  1549.957  1549.95700  1680.8205   

                             
             75%        max  
year                         
1981  6079.24400   6079.244  
1985  6178.52400   6178.524  
1989  7229.12000   7229.120  
1990  4706.34600   6155.646  
1991  2774.25500   6110.719  
1992  1559.23300   2990.413  
1993  2870.98100   5804.319  
1994  3356.09800   8018.721  
1995  4476.79525   6584.739  
1996  3886.13175   7770.539  
1997  2119.91800   6445.771  
1998  2272.80950   6590.262  
1999  4037.45400   4798.447  
2000  3195.39400  11762.945  
2001  3796.67800   7805.312  
2002  3795.41100   8580.888  
2003  1994.81500   9321.219  
2004  3061.82800   7788.167  
2005  3880.01400   8606.334  
2006  3691.05000   9142.805  
2007  2336.08500   5308.632  
2008  2233.33875   8451.479  
2009  5224.83800   9837.877  
2010  2235.32300   8979.332  
2011  4218.38200  10559.206  
2012  5748.59100  12079.545  
2013  4987.80500  12695.235  
2014  3489.86400   6547.308  
2015  5924.00875  10510.771  
2016  2575.45600   9190.003  
2017  3338.44200   4249.804  
2018  4539.08700  12573.956  
2019  6567.91000  28693.063  
2020  1811.68400   1811.684

In [80]:
data_tidy.loc[:, ['year','cpi2010']].groupby('year').describe()

cpi2010                                                                   \
       count        mean         std      min        25%       50%        75%   
year                                                                            
1981    36.0    0.000000    0.000000    0.000    0.00000    0.0000    0.00000   
1985    36.0    0.000000    0.000000    0.000    0.00000    0.0000    0.00000   
1989   108.0   18.569667   14.710811    7.186    7.18600    9.2810   39.24200   
1990   288.0   21.493625   17.910186    0.001    7.98200   15.7050   34.40925   
1991   360.0   28.520600   22.569267    0.005   12.08100   16.6860   46.97800   
1992   360.0   29.126200   22.191499    1.139    2.92000   27.6335   49.22800   
1993   324.0   22.434000   14.946775    0.336   11.15200   19.3500   36.64600   
1994   396.0   37.190182   23.721174    2.024   14.67000   51.0940   59.44800   
1995   576.0   37.650938   18.794739    6.906   26.96275   35.0170   42.58550   
1996   612.0   39.818824   16.785738    9.630   24.93400   42.4140   43.76000   
1997   468.0   48.615000   22.638109    8.946   28.90400   51.4890   68.75400   
1998   792.0   52.640545   22.074029    8.064   35.23500   54.0630   73.17100   
1999   396.0   38.298091   19.891072    0.684   23.25100   38.6900   52.74400   
2000  1908.0   54.083358   16.987704   14.991   41.94300   51.8590   72.16900   
2001   792.0   52.120636   18.854030    7.347   40.73700   49.5890   62.85500   
2002   648.0   62.467000   14.822294   35.357   49.71500   62.1895   79.15900   
2003   648.0   64.106333   17.224515   38.576   45.29200   64.3565   82.01300   
2004   540.0   68.503533   12.230820   48.276   55.62500   68.3480   81.94400   
2005  1440.0   70.442000   12.352195   42.234   61.58700   68.6820   81.34475   
2006  1296.0   78.624306    9.786222   58.706   72.74350   80.6445   87.29900   
2007   720.0   78.920850    7.494781   63.006   73.84600   79.5075   83.25325   
2008   576.0   85.817250    7.121045   75.438   79.84350   85.7530   89.59900   
2009   648.0   95.647611    2.116132   90.871   94.20600   95.5315   96.65400   
2010  1116.0  100.000000    0.000000  100.000  100.00000  100.0000  100.00000   
2011   864.0  108.215500    5.945956  101.760  104.90350  106.7750  109.63025   
2012   864.0  113.183083    7.896413  103.411  107.51975  111.6690  118.48775   
2013   576.0  124.282375   14.056965  108.172  115.36200  120.7240  129.39375   
2014   900.0  130.094760   37.399815  104.463  109.92000  118.9500  137.42100   
2015   504.0  137.673214   35.322247  106.213  117.59000  127.0445  147.86500   
2016   468.0  153.511692   31.150921  111.686  132.70600  144.0150  165.78400   
2017   288.0  148.858500   40.610046  110.792  118.92300  135.5250  161.62900   
2018   720.0  149.026750   39.381330  102.391  116.19200  146.6805  164.57275   
2019   612.0  193.683588  111.757535  109.251  117.69800  150.3440  188.73000   
2020    72.0  161.284500   23.206218  138.240  138.24000  161.2845  184.32900   

               
          max  
year           
1981    0.000  
1985    0.000  
1989   39.242  
1990   49.759  
1991   69.721  
1992   64.976  
1993   39.528  
1994   61.996  
1995   72.937  
1996   65.004  
1997   82.474  
1998   81.536  
1999   75.144  
2000   81.588  
2001   83.290  
2002   82.382  
2003   87.532  
2004   85.740  
2005   88.747  
2006   90.585  
2007   90.968  
2008   97.791  
2009  100.770  
2010  100.000  
2011  132.015  
2012  139.826  
2013  166.125  
2014  299.223  
2015  250.619  
2016  218.586  
2017  241.062  
2018  240.201  
2019  508.339  
2020  184.329

#### `prop` summary grouped by years.

In [70]:
data_tidy.loc[:, ['year','prop']].groupby(['year']).describe()

prop                                                            
       count      mean       std    min      25%     50%      75%    max
year                                                                    
1981    36.0  0.465500  0.301750  0.028  0.17800  0.4690  0.72250  0.938
1985    36.0  0.481528  0.308450  0.021  0.18100  0.4995  0.76725  0.947
1989   108.0  0.785361  0.218442  0.158  0.70650  0.8855  0.94300  0.995
1990   252.0  0.740282  0.283810  0.030  0.58450  0.8730  0.97400  1.000
1991   360.0  0.649672  0.291056  0.016  0.43425  0.7470  0.89850  0.996
1992   504.0  0.589631  0.299883  0.002  0.33025  0.6495  0.84500  0.995
1993   396.0  0.695737  0.269002  0.022  0.49575  0.7725  0.93700  0.998
1994   468.0  0.612859  0.335876  0.000  0.31000  0.6990  0.94200  1.000
1995   576.0  0.687719  0.305519  0.003  0.45850  0.8050  0.95300  1.000
1996   756.0  0.634366  0.298583  0.000  0.39950  0.7130  0.89650  1.000
1997   504.0  0.675308  0.316310  0.001  0.44650  0.7990  0.95200  1.000
1998   864.0  0.644363  0.298727  0.001  0.42775  0.7330  0.91125  1.000
1999   468.0  0.688893  0.296362  0.004  0.48025  0.7910  0.95025  1.000
2000  2052.0  0.661070  0.303410  0.000  0.43100  0.7550  0.93425  1.000
2001   792.0  0.617811  0.293460  0.002  0.39250  0.6850  0.88325  0.994
2002   648.0  0.805924  0.243386  0.016  0.71750  0.9280  0.97725  1.000
2003   648.0  0.642017  0.280604  0.006  0.43925  0.7095  0.88900  0.999
2004   540.0  0.673076  0.277672  0.009  0.46300  0.7490  0.92000  1.000
2005  1512.0  0.698636  0.289233  0.009  0.49550  0.8030  0.95500  1.000
2006  1332.0  0.736251  0.271396  0.002  0.56275  0.8460  0.96200  1.000
2007   684.0  0.703917  0.275387  0.009  0.52975  0.8025  0.93800  1.000
2008   576.0  0.703457  0.268226  0.017  0.52975  0.7965  0.93100  0.996
2009   648.0  0.838086  0.212645  0.020  0.78025  0.9340  0.98300  1.000
2010  1116.0  0.702295  0.279653  0.011  0.50775  0.7940  0.94925  1.000
2011   900.0  0.748767  0.249004  0.017  0.60200  0.8495  0.95200  1.000
2012   864.0  0.780125  0.242255  0.018  0.65400  0.8810  0.97400  1.000
2013   576.0  0.771319  0.253725  0.003  0.64875  0.8815  0.96625  1.000
2014   900.0  0.781441  0.242638  0.019  0.65500  0.8870  0.97100  1.000
2015   576.0  0.807691  0.229892  0.034  0.71775  0.9040  0.98400  1.000
2016   468.0  0.739562  0.243781  0.063  0.57375  0.8195  0.94600  1.000
2017   324.0  0.807466  0.212154  0.106  0.69300  0.8930  0.97325  0.997
2018   792.0  0.783139  0.251244  0.008  0.65175  0.9045  0.97900  1.000
2019   828.0  0.827772  0.229295  0.036  0.74200  0.9450  0.99100  1.000
2020    72.0  0.725097  0.260631  0.070  0.60100  0.8270  0.93725  0.976

#### `prop` summary grouped by level

In [69]:
data_tidy.loc[:, ['level','prop']].groupby(['level']).describe()

prop                                                            
        count      mean       std    min      25%     50%      75%    max
level                                                                    
1      2464.0  0.886287  0.167455  0.011  0.85975  0.9595  0.98700  1.000
2      2464.0  0.874638  0.172608  0.011  0.83800  0.9490  0.98300  1.000
3      2464.0  0.851898  0.183777  0.011  0.79700  0.9290  0.97600  1.000
4      2464.0  0.817203  0.200569  0.010  0.73500  0.8950  0.96425  1.000
5      2464.0  0.769976  0.223202  0.008  0.65700  0.8480  0.94700  1.000
6      2464.0  0.700446  0.254869  0.007  0.53075  0.7680  0.91425  1.000
7      2464.0  0.603727  0.276006  0.004  0.39000  0.6380  0.83900  1.000
8      2464.0  0.513729  0.287668  0.000  0.26875  0.5125  0.75200  1.000
9      2464.0  0.401914  0.269362  0.000  0.16300  0.3730  0.61425  0.989

Looking at the means of our summary tables, there is clearly some sort of trend. We will explore a little bit about these trends in the exploratory analysis below.

### Exploratory analysis

If you were lucky and your dataset was neat, you should aim to include a few exploratory plots or tables here -- they don't need to be polished at this stage, but you should select plots that are informative (rather than including all plots you may have looked at). 

If you do include exploratory graphics or tables, please explain in a sentence or two what each one shows. Try to include a minimum of code. Consider [saving your plots as images](https://altair-viz.github.io/user_guide/saving_charts.html#png-svg-and-pdf-format) and inputting images into markdown cells instead of generating them anew via code cells.

In [74]:
alt.data_transformers.enable('default', max_rows=None)
from visualize import line_plot1, line_plot2, line_plot3, line_plot4

### Visualize Trends

We won't do any complex EDA here, but we will take a look at some trends we saw in the data summary with some simple line charts.

#### 1) `Average CPI` against `year`

In [75]:
line_plot1(data_tidy)

alt.Chart(...)

#### 2) `Average prop of kids that completed grades(1-9)` against `year`

The line graph below shows an upwards trend in `prop` of kids that attained a grade as `year` increases.

In [76]:
line_plot2(data_tidy)

alt.Chart(...)

The natural followup question to plot-2 is whether `sex` and `location` of the kids affect the `prop`.

Below are 2 plots of plot-2 faceted by `sex` and `location` respectively.

#### 3) `plot2` faceted by `sex`

In [77]:
line_plot3(data_tidy)

alt.FacetChart(...)

#### 4) `plot2` faeted by `location`

In [78]:
line_plot4(data_tidy)

alt.FacetChart(...)

---
## 3. Planned work

Here you should indicate your tentative ideas for your analysis. Don't worry, these aren't final -- you can always change your mind later or shift gears if they don't pan out. The objective is to have you start thinking ahead about what you'll do.

### Questions

1. Are there significant differences in educational attainment depending on wealth?
2. Do gender and living standards affect a person's educational attainment? 

### Proposed approaches

1. For our first question we must explore the relationship between educational attainment and wealth. To explore this relationship it will be useful to first explore the distribution of each variable on its own. We would explore educational attainment by plotting against the observed proportion to see the proportion of the sample that attained the education at each grade level across all countries combined. We can then build a seperate scatterplot of our variables prop vs. gpdpc2015 where we can encode the points within the scatterplot to represent another variable such as the countries to give us more information about our data. 

2. For our second question we are looking into how gender and living standards affect educational attainment. For this question we would explore the relationship of these variables by plotting educational attainment vs year. We would encode each plot to show the proportion of attained education with a different line for each location (whether rural or urban) and be faceted by sex to allow us to have a more clear visual of any significant trends that are attributed to either sex or location. 

---
## Submission Checklist
1. Save file to confirm all changes are on disk
2. Run *Kernel > Restart & Run All* to execute all code from top to bottom
3. Save file again to write any new output to disk
4. Generate PDF and submit to Gradescope